### DISCUSSION TITLE
Installation zusammen mit ioBroker auf einem RaspberryPi (ohne Virtualisierung)

### DISCUSSION THREAD

**Discussion by DominikErnst** [OpenCCU/OpenCCU#1167] [🇩🇪 Fragen&Antworten] [answered]
Hallo, ich möchte auf einem Raspberry 4 IOBroker und RaspberryMatic/CCU3 parallel laufen lassen und das ohne diesem nutzlosen und komplizierten Virtualisierunsgedöns.

Ist das möglich?
Wie installiere ich RaspberryMatic manuell?
IOBroker lässt sich "nur" noch über curl installieren, gibt es so ein script auch für RaspberryMatic?

Das ganze sollte natürlich auf dem original Raspi OS laufen.

Vielen Dank.

**Comment by jens-maus** [comment, marked as answer]
> Hallo, ich möchte auf einem Raspberry 4 IOBroker und RaspberryMatic/CCU3 parallel laufen lassen und das ohne diesem nutzlosen und komplizierten Virtualisierunsgedöns.

Schön wäre es doch, wenn man solch eine Frage etwas wertungsfreier verfassen könnte, oder? Nur weil für einen selbst Virtualisierungslösungen "nutzlos" und "kompliziert" erscheinen, muss das nicht für die Mehrheit zutreffen.

> Ist das möglich?
> Wie installiere ich RaspberryMatic manuell?
> IOBroker lässt sich "nur" noch über curl installieren, gibt es so ein script auch für RaspberryMatic?

Es sei hier auf die umfangreiche Dokumentation zu RaspberryMatic hingewiesen (https://github.com/jens-maus/RaspberryMatic/wiki). Dort sollte man alle notwendigen Informationen zu möglichen Installationsszenarien auf verschiedenen Hardwareplatformen und Umgebungen herauslesen können.

> Das ganze sollte natürlich auf dem original Raspi OS laufen.

IOBroker und RaspberryMatic zu vergleichen ist hier leider etwas zu kurz gedacht. IOBroker ist eine Middleware, RaspberryMatic ansich ein vollwertiges Betriebssystem samt CCU Umgebung. Und genauso wie man unter Linux kein Windows via "curl" installieren kann, so geht das eben auch mit RaspberryMatic nicht. Genau dafür existiert jedoch die Möglichkeit (wenn man eine RaspberryMatic CCU eben doch unter RaspberryPi OS betreiben will) RaspberryMatic entweder als virtuelle OVA Maschinen unter VirtualBox&Co zu installieren oder eben seit neuestem eine Docker Umgebung einzusetzen. Wem das zu kompliziert ist, der greife eben zu anderen Lösungen (z.B. [piVCCU](https://github.com/alexreinert/piVCCU) oder [debmatic](https://github.com/alexreinert/debmatic)) oder kaufe eine Standard CCU3 von eQ3/ELV.

**Reply by DominikErnst** [reply]
Erstmal vielen Dank, für die schnelle Antwort. 
Grundsätzlich hast du Recht mit "wertfrei", ich bin aber von dem ganzen Zeug sehr enttäuscht. Es macht keinen Sinn das zu virtualisieren, man verliert nur Performance und einfache Handhabung, AFAIK die Funkmodule könne nicht angesprochen werden!??! Egal was man sich anschaut CCU, IOBroker, Flot,  ... es ist veraltete Software aus dem letzten Jahrtausend, deswegen meine Frustration die ich hier rausgelassen habe. SORRY!
Ich bin selber seit 30 Jahren beruflich (40 privat) Jahren Softwareentwickler, das kann man alles moderner, schöner und vor allem einfacher machen. Ok, vieles ist kostenlos, aber aus dem Bastelalter müsste das mal raus kommen.
Zurück zu Thema.
Habe zur manuelle Installation mehrere Stunden gegoogelt/gelesen und nichts gefunden, leider auch nicht unter dem angegebenen Link.
Interessant ist deine Aussage, dass RaspberryMatic (fast) ein OS ist. (Ich würde die CCU ja eher als App entwickeln, aber wahrscheinlich ist das "historisch so gewachsen" ;-))
Noch eine CCU kaufen kommt nicht in Frage, da ich mein Haus nicht zumüllen will, vom Stromverbrauch mal abgesehen, eine Tahoma steht ja auch schon rum. (Übrigens auch überteuerter Müll, da ich seit einem Jahr versuche noch einen Rollladen und eine Markise einzubinden.) Das ist ja eben auch gerade meine Frage nämlich den IOBroker und CCU3 auf einem Raspi zum Laufen bringen.
Unter Windows lässt sich übrigens Linux installieren WSL2 mit "dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart". Wie auch immer das dann auf ..äh.. unter Windows läuft ...
Habe schon viel ausprobiert und jetzt alle drei Funkadapter hier liegen, da die Unterstützung bei jeder möglichen Lösung unterschiedlich war. #Frust

Kann ich dann RaspberryMatic installieren und da IOBroker dazu installieren oder empfiehlst du eher einen Raspi aufsetzen, dann mit Docker den RaspberryMatic laufen lassen und auf dem Haupt-OS einfach IOBroker installieren?
Läuft dann mit der Docker Lösung auch .. ah gerade gelesen, Stichwort "Kernel-Driver" .... oje naja ich hoffe, das ich da eine Anleitung finde....

Vielen Dank
PS: Hab' dir gerade #1167 geschickt. Danke für dein Arbeit.
